fix: change gateway

jrainville opened the pull request:
Tempoary fix to the IPFS gateway not working.

It's slow, it has a limit and it bugs sometimes, but at least it works better than nothing at all.

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 9532c97a | [#1](https://ci.status.im/job/status-desktop/job/branches/job/macos/job/PR-6965/1/) | 2022-08-10 19:43:35 | ~8 min | `macos` | [:package:`dmg`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Desktop-220810-193453-9532c9-pr6965.dmg) |

jrainville commented:
Re-asking for review, because I changed the gateway again to the status one instead and I also removed all the old code that decoded the hashes. Instead, I just use the url that comes from status-go.

jrainville commented:
Status-go PR https://github.com/status-im/status-go/pull/2795

caybro reviewed (approved):
Looks good!

richard-ramos reviewed (approved):
:clap:

jrainville commented:
I updated the Makefile and status-go code so that now we pass the build mode. In dev, we use the status gateway, in prod, we use the infura one. 

Please re-test and re-review.

jakubgs commented on Makefile:
```diff
@@ -390,7 +391,8 @@ $(FCITX5_QT): | deps
 
-$(STATUS_CLIENT_APPIMAGE): override RESOURCES_LAYOUT := $(PRODUCTION_PARAMETERS)
+$(STATUS_CLIENT_APPIMAGE): override RESOURCES_LAYOUT := $(PRODUCTION_PARAMETERS) \
+	MODE_ARG := PROD
```
Pretty sure you can't break `Makefile` lines like that.
```suggestion
$(STATUS_CLIENT_APPIMAGE): MODE_ARG := PROD
```

anastasiyaig commented:
https://github.com/status-im/status-desktop/pull/6965/files#r951366006 Jakub says the builds are failing because of that @jrainville

jrainville commented:
Updated. It should build now

jrainville commented:
Investigating why it still doesn't build

jrainville commented:
Ok, now I updated correctly. I'm not sure why my previous push didn't push the changes correctly.

jakubgs commented on Makefile:
```diff
@@ -187,6 +187,8 @@ ifeq ($(detected_OS),Darwin)
 endif
 
+IPFS_GATEWAY_URL_ARG := https://ipfs.status.im/
+PROD_IPFS_GATEWAY_URL_ARG := https://status-im.infura-ipfs.io/ipfs/
```
Are you sure you want to use `:=` and make it not overridable?

jakubgs replied:
If you want env variables to override those use lazy assignment: `?=`

Also, it seems to me like `IPFS_GATEWAY_URL_ARG` is not actually used since it's always replaced by `PROD_IPFS_GATEWAY_URL_ARG`. Wouldn't it make more sense and call those `IPFS_GATEWAY_URL_ARG_DEV` and `IPFS_GATEWAY_URL_ARG_PROD`?

jakubgs commented on Makefile:
```diff
@@ -391,6 +393,7 @@ $(FCITX5_QT): | deps
 PRODUCTION_PARAMETERS := -d:production -d:chronicles_sinks=textlines[stdout],textlines[nocolors,dynamic]
 
 $(STATUS_CLIENT_APPIMAGE): override RESOURCES_LAYOUT := $(PRODUCTION_PARAMETERS)
+$(STATUS_CLIENT_APPIMAGE): override IPFS_GATEWAY_URL_ARG := $(PROD_IPFS_GATEWAY_URL_ARG)
```
Any specific reason why we are using the [`override` directive](https://www.gnu.org/software/make/manual/html_node/Override-Directive.html) here? Or are we just cargo-culting it?

jrainville replied:
![image](https://user-images.githubusercontent.com/11926403/186230834-16d4d6ac-3cd5-4084-9571-f2933a59cd35.png)

jakubgs commented on Makefile:
```diff
@@ -391,6 +393,7 @@ $(FCITX5_QT): | deps
 PRODUCTION_PARAMETERS := -d:production -d:chronicles_sinks=textlines[stdout],textlines[nocolors,dynamic]
 
 $(STATUS_CLIENT_APPIMAGE): override RESOURCES_LAYOUT := $(PRODUCTION_PARAMETERS)
+$(STATUS_CLIENT_APPIMAGE): IPFS_GATEWAY_URL_ARG = $(PROD_IPFS_GATEWAY_URL_ARG)
```
Why are you using the prod endpoint for non-release builds?

jrainville replied:
As far as I understand, those are when we do `make pkg`, so it's what we do when we create a release build? Which command would be the one that is used for release otherwise?
I followed the places where we use `PRODUCTION_PARAMETERS`

jakubgs replied:
The way you are doing it it will affect all builds. The correct way is detecting a release build in CI. Like I do here:

* https://github.com/status-im/status-jenkins-lib/pull/44
* https://github.com/status-im/status-mobile/pull/13850

I use [`utils.isReleaseBuild()`](https://github.com/status-im/status-jenkins-lib/blob/db0161b8d79493f7d7b14f8d736e2b162f8244d8/vars/utils.groovy#L151-L159) to provide a different URL for actual release builds.

jakubgs replied:
Unless your intention isn't to actually use that URL for actual release builds but for CI builds as a whole.

Can you clarify the intention? I assumed that we want to distinguish between actual release builds like in mobile.

jrainville replied:
I think the goal would be mostly for release builds, but I don't know how to differentiate a normal package build vs a release build. 
We have `RELEASE ?= false`, but it never gets set to `true`.

Anyway, since this should be a temporary solution, I think using the "prod" url on all builds makes sense, since QAs use those builds to test the app and if we ever need a build to pass to a test user, we'd want the prod url as well I think.

What do you think?

jakubgs replied:
Honestly, none of this `IPFS_GATEWAY_URL_ARG` is necessary at all since you've used `?=` assignment in your PR.

That means we can entirely drop any of the `Makefile` changes here, and just do it in `status-jenkins-lib`, like this:
https://github.com/status-im/status-jenkins-lib/commit/d4e69880
I made that change part of my PR intended initially for mobile:

* https://github.com/status-im/status-jenkins-lib/pull/44

jrainville replied:
Ah cool, so as I understand, with that change you did here https://github.com/status-im/status-jenkins-lib/commit/d4e69880, I can drop the changes in the Makefle?

jakubgs replied:
Correct, this way the only builds that will be affected by a different URL are actually release builds based on the `utils.isReleaseBuild()` check.

jrainville replied:
Updated. No more Makefile changes

jakubgs replied:
But you are not using the changes in `status-jenkins-lib`.

jrainville replied:
Updated

jakubgs commented on Makefile:
```diff
@@ -455,6 +458,7 @@ MACOS_INNER_BUNDLE := $(MACOS_OUTER_BUNDLE)/Contents/Frameworks/QtWebEngineCore.
 STATUS_CLIENT_DMG ?= pkg/Status.dmg
 
 $(STATUS_CLIENT_DMG): override RESOURCES_LAYOUT := $(PRODUCTION_PARAMETERS)
+$(STATUS_CLIENT_DMG): IPFS_GATEWAY_URL_ARG = $(PROD_IPFS_GATEWAY_URL_ARG)
```
Same here, prod endpoint for non-release builds.

jakubgs commented on Makefile:
```diff
@@ -527,6 +531,7 @@ STATUS_CLIENT_EXE ?= pkg/Status.exe
 STATUS_CLIENT_7Z ?= pkg/Status.7z
 
 $(STATUS_CLIENT_EXE): override RESOURCES_LAYOUT := $(PRODUCTION_PARAMETERS)
+$(STATUS_CLIENT_EXE): IPFS_GATEWAY_URL_ARG = $(PROD_IPFS_GATEWAY_URL_ARG)
```
Same.
